### ISSUE TITLE
Introduce a tickler type to organize date specific triggers

### ISSUE THREAD

**Issue by doolio** [Trevoke/org-gtd.el#176]
I propose we introduce a tickler type for date specific triggers. This is for items we wish to be reminded of on a specific date-time. The assigned date-time could be an active date-time or a scheduled date-time. If an active date-time we simply just want to be tickled/triggered on that date about the item but with a scheduled date-time we have decided we want to start the project/task on that specific date. Note, scheduled date-time items will remain in the agenda view until action is taken upon them whereas non-scheduled items but with an active date-time will appear in the agenda view only on the assigned date-time. An active date-time can appear anywhere in a headline but we could introduce a `:TICKLE:` property to sit right  below  the headline à la `:SCHEDULED:`. Otherwise, I propose we simply put the date-time stamp in the headline itself.

In terms of organizing (step 3) such items, I propose we use two separate actions `tickle` and `schedule` (and keep the `incubate` action (see #175) for organizing items to a someday/maybe file where we don't assign a date-time). Therefore, `org-gtd-tickle` could be used to assign an active date-time to an item and then move it to a tickler file e.g. `tickler.org`. And `org-gtd-schedule` could be used to assign a scheduled date-time and then move it to either a calendar file e.g. `calendar.org` or a tickler file `tickler.org`. I propose the latter and then use a `calendar.org` file for date-time events that don't shift i.e. anniversaries, birthdays, national holidays etc.

For the tickler file, I propose we create 43 (31+12) top level headlines - one for each possible day of a month (simply named 1 to 31) and one for each month of the year (simply named January to December). Though not necessary with Org as no matter where an item appears in the file it will appear in the agenda view, these headlines would serve to provide a clearer overview of date-time events across the year or years if the user were to view the file directly.

**Comment by Trevoke** [comment]
The core value of org-mode when used as org-gtd does is that the agenda view is the one way to think about interacting with the tasks.

It's unclear to me what the difference is, in what you suggested, between:
- physically opening the tickler file
- physically opening a someday/maybe file

It feels like the only difference between those two is "the user knows where to go on a given day/month", which then means that opening the agenda view can serve exactly the same purpose for the user.

With more words, more specifically for the 43 headings:

The reason I didn't create a tickler file as you suggested is because I didn't think it was valuable to implement a digital tool based on the same restrictions provided by a physical tool. The agenda is the entry point for viewing these files. If we create a tickler file that has the 43 headlines, then it's clear the intent is for this to be viewed manually by the user, and therefore the agenda view won't actually provide of anything of value, so there's no point in implementing one.

It does seem like we miss some sort of "I'll review this at some point but I don't know when" so I do think there's room to introduce a "someday/maybe" where information will go to die if the user doesn't open the file themselves, though, for big, random, vague ideas, that we want to review together because we want to compare them to one another, like "buy a boat" or "move to Hawaii".

**Comment by doolio** [comment]
> The core value of org-mode when used as org-gtd does is that the agenda view is the one way to think about interacting with the tasks.

You are right of course and it is primarily how I interact with the files in my GTD system. However, I currently have items in my `tickler.org` file for reminder many years from now some as much as ten years from now e.g. renewal of passports. Sometimes, I like to get an overview of such items and I do so by looking at the tickler file directly. I don't believe you can make a specific agenda view for such an overview but perhaps you know a way? And because I on occasion want such an overview I've found it useful to have the 43 headlines for clarity. But I agree it is not necessary.

> so I do think there's room to introduce a "someday/maybe" where information will go to die if the user doesn't open the file themselves

If I'm not mistaken David Allen does suggest one review their someday/maybe list during their weekly review.

**Comment by Trevoke** [comment]
When you say "such items", do you just mean "the ones far in the future" ? Because it's really easy to create an agenda view that only shows things that have a timestamp for over a year in the future.. Or, really, greater than any arbitrary date, or even within a range.

Re: the weekly review, I think you're right, although what David Allen suggests and what people do are often two very different things :)
